### PR TITLE
dendrite: 0.5.1 -> 0.6.5

### PR DESCRIPTION
**Pull request by tim-tx**
###### Motivation for this change
dendrite-0.6.5 release

https://github.com/matrix-org/dendrite/releases/tag/v0.6.5

###### Things done

Added new config options for database URLs, these are now required to be present in the config.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [x] [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - [ ] and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - [ ] or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - [ ] made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tim-tx** [comment]
This still needs work.

**Comment by genofire** [comment]
0.6.4 is released

**Comment by NickCao** [comment]
Running fine on my middle sized homeserver, push notification Indeed works. Just need a rebase for the conflict in release note.

**Comment by tim-tx** [comment]
I updated the PR to 0.6.5. I had thought of reorganizing the config, but I thought I might be too tempted to over-engineer it, and I am a first-time contributor. As of now the derivation maps a Nix attribute set directly to the dendrite yaml (as this derivation has always done). I added explicit (required & with defaults) options for db connection string urls, with defaults set to split sqlite databases. Upstream recommends postgres (single db) in production. The config then seems to be asking for some reduction:
```
services.dendrite = {
  enable = true;
  settings = let url = "postgres:///dendrite?host=/run/postgresql"; in {
    app_service_api.database.connection_string = url;
    media_api.database.connection_string = url;
    federation_api.database.connection_string = url;
    key_server.database.connection_string = url;
    room_server.database.connection_string = url;
    sync_api.database.connection_string = url;
    user_api.account_database.connection_string = url;
    user_api.device_database.connection_string = url;
    mscs.database.connection_string = url;
    global = {
      server_name = "example.com";
      private_key = "/var/lib/dendrite/matrix_key.pem";
    };
  };
};
```
But I will leave it for another contributor or another day.

I guess we are just waiting on review from someone on the maintainer list. Merge conflicts should be fixed

**Review by chuangzhu — APPROVED** [review]
Result of `nixpkgs-review pr 157923` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dendrite</li>
  </ul>
</details>

**Comment by genofire** [comment]
Does it backported to 21.11 ?

**Comment by McSinyx** [comment]
> connection string urls [...] The config then seems to be asking for some reduction

FWIW [upstream now supports `global.connection_string`](https://matrix-org.github.io/dendrite/installation/configuration#global-connection-pool-monolith-with-a-single-postgresql-database-only).

**Pull request merged**
